### PR TITLE
Add canonical URL conversion support.

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: n/a

Based on: https://github.com/moderntribe/tribe-common/pull/1027

This PR implements, and tests, the base code implemented in `common` to convert "ugly" links to their canonical, probably prettier, version.
An example:

```php
$ugly = home_ur( '/post_type=tribe_events&eventDisplay=list&paged=2' );

// Will return /events/list/page/2
$pretty = Tribe__Events__Rewrite::instance()->get_canonical_url( $ugly );
```

### CODE REVIEW

File: tests/wpunit/Tribe/Events/RewriteTest.php [hunk 1]
```diff
@@ -56,4 +57,91 @@ public function it_should_filter_post_type_link_for_supported_post_types_only()
 
 		$this->assertEquals( 'foo', $sut->filter_post_type_link( 'foo', $post ) );
 	}
+
+	private function make_instance() {
+		return new Rewrite( $this->wp_rewrite->reveal() );
+	}
+
+	public function canonical_urls(  ) {
```
Comment (bordoni) [inline tests/wpunit/Tribe/Events/RewriteTest.php]: Spacing on `()`

File: tests/wpunit/Tribe/Events/RewriteTest.php [hunk 2]
```diff
@@ -56,4 +57,91 @@ public function it_should_filter_post_type_link_for_supported_post_types_only()
...
+	 * It should allow converting a URL to its canonical form
+	 *
+	 * @test
+	 * @dataProvider canonical_urls
+	 */
+	public function should_allow_converting_a_url_to_its_canonical_form($uri, $expected) {
+		global $wp_rewrite;
+		$wp_rewrite->permalink_structure = '/%postname%/';
+		$wp_rewrite->rewrite_rules();
+		$canonical_url = ( new Rewrite )->get_canonical_url( home_url($uri) );
```
Comment (bordoni) [inline tests/wpunit/Tribe/Events/RewriteTest.php]: Spacing around `()`

File: tests/wpunit/Tribe/Events/RewriteTest.php [hunk 3]
```diff
@@ -56,4 +57,91 @@ public function it_should_filter_post_type_link_for_supported_post_types_only()
...
+			],
+		];
+}
+	/**
+	 * It should allow converting a URL to its canonical form
+	 *
+	 * @test
+	 * @dataProvider canonical_urls
+	 */
+	public function should_allow_converting_a_url_to_its_canonical_form($uri, $expected) {
```
Comment (bordoni) [inline tests/wpunit/Tribe/Events/RewriteTest.php]: Spacing around `()`

File: tests/wpunit/Tribe/Events/RewriteTest.php [hunk 4]
```diff
@@ -56,4 +57,91 @@ public function it_should_filter_post_type_link_for_supported_post_types_only()
...
+		global $wp_rewrite;
+		$wp_rewrite->permalink_structure = '/%postname%/';
```
Comment (bordoni) [inline tests/wpunit/Tribe/Events/RewriteTest.php]: Should we test with a couple of other Permastructures?

Comment (lucatume) [reply tests/wpunit/Tribe/Events/RewriteTest.php]: It's really indifferent here, we're generating the rules the same way.

**Pull request merged**